java/cdk/src/main/java/sleeper/cdk/stack/GarbageCollectorStack.java

```diff
@@ -77,10 +78,16 @@ public GarbageCollectorStack(
         String triggerFunctionName = String.join("-", "sleeper", instanceId, "garbage-collector-trigger");
         String functionName = String.join("-", "sleeper", instanceId, "garbage-collector");
 
-        // Timeout is set to 90% of the period with which this runs to avoid 2 running simultaneously,
-        // with a maximum of 900 seconds (15 minutes) which is the maximum execution time
-        // of a lambda.
-        int timeoutSecsFromGcPeriod = (int) (0.9 * 60 * instanceProperties.getInt(GARBAGE_COLLECTOR_PERIOD_IN_MINUTES));
+        // Timeout is set to either the configured timeout period within the properties or defaults to the period for which this runs.
+        // Both have a maximum of 900 seconds (15 minutes) which is the maximum execution time of a lambda.
+        int timeoutSecsFromGcPeriod = -1;
+        int gcPeriodTimeout = instanceProperties.getInt(GARBAGE_COLLECTOR_LAMBDA_TIMEOUT_IN_MINUTES);
+        if (gcPeriodTimeout > 0) {
+            timeoutSecsFromGcPeriod = (60 * gcPeriodTimeout);
+        } else {
+            timeoutSecsFromGcPeriod = (60 * instanceProperties.getInt(GARBAGE_COLLECTOR_PERIOD_IN_MINUTES));
+        }
+
         Duration handlerTimeout = Duration.seconds(Math.max(1, Math.min(timeoutSecsFromGcPeriod, 900)));
```

Comment:
It looks like we can remove this line? We could add validation directly on the property to keep it in the range that it's being truncated to here?

Have a look at the validation predicate on GARBAGE_COLLECTOR_TABLE_BATCH_SIZE.